### ISSUE TITLE
GPU support?

### ISSUE THREAD

**Issue by mirfan899** [persephone-tools/persephone#236]
I have tried to run it with GPU but it does not work. Using TensorFlow-GPU shows import error and asks to install TensorFlow only.

![cpu](https://user-images.githubusercontent.com/3822565/97794921-cf28fb00-1c21-11eb-86a3-764c157f3903.png)
![gpu](https://user-images.githubusercontent.com/3822565/97794922-d18b5500-1c21-11eb-8fdb-f49117d4fe2c.png)

How to train it with GPU? I read the documentation but it seems GPU support is missing.

**Comment by oadams** [comment]
The import error is probably what needs more investigation. The code does support CUDA. Perhaps the issue is that you have CUDA 11.0 installed and the version of tensorflow used by persephone (which is years old at this point) doesn't support that recent version of CUDA.

**Comment by mirfan899** [comment]
Yes, it seems I need Cuda 9 to use TensorFlow-GPU 1.15.4

**Comment by shuttle1987** [comment]
Could you please provide us with the exact import error text?

**Comment by mirfan899** [comment]
My Cuda version is 10.1 and it throws an error of Tensorflow GPU not installed.

Okay, what after googling the issue I have found out that Cuda 10.1 does not support TensorFlow-GPU==1.15.4. I need to downgrade the Cuda or use TensorFlow GPU > 2.1

https://www.tensorflow.org/install/source#gpu

**Comment by mirfan899** [comment]
Okay, another thing is if you install tensorflow-gpu==1.15.0, it works.

**Comment by shuttle1987** [comment]
Sounds like a versioning mismatch issue. Off the top of my head I'm not entirely sure if we can bump to `tensorflow-gpu>2` for the dependency in this library, but if it works for you then that's great.

@oadams thoughts about bumping versions of tensorflow?

**Comment by oadams** [comment]
Yeah we can bump the version to something that supports the newer versions of CUDA. If 1.15.0 works for you @mirfan899 then maybe we go with that.
